templates/nginx-cfg.j2

```diff
@@ -22,4 +22,17 @@ server {
     proxy_redirect http:// https://;
     proxy_pass              http://jenkins;
   }
+
+  # Restrict the following paths to localhost (SSH tunneled users only)
+  location ~ /(api|restart|quietDown|user/jenkins|jnlpJars|script|asynchPeople|credential-store|systemInfo|view/All/newJob|) {
+    deny    all;
+    allow   localhost;
```

Comment:
An ssh tunnel could allow bypassing bypassing nginx entirely with a direct connection to jenkins on port 8080, so we could leave this out if we wanted to.